mplc/multi_partner_learning.py

```diff
@@ -110,28 +127,50 @@ def save_final_model(self):
         model_to_save = self.build_model()
         model_to_save.save_weights(os.path.join(model_folder, self.dataset_name + '_final_weights.h5'))
 
-    def split_in_minibatches(self):
-        """Split the dataset passed as argument in mini-batches"""
+    def log_partner_perf(self, partner_id, partner_index, history):
+        for key_history in self.history.metrics:
+            self.history.history[partner_id][key_history][self.epoch_index,
+                                                          self.minibatch_index] = history[key_history][-1]
 
-        for partner in self.partners_list:
-            partner.split_minibatches()
+        epoch_nb_str = f"Epoch {str(self.epoch_index).zfill(2)}/{str(self.epoch_count - 1).zfill(2)}"
+        mb_nb_str = f"Minibatch {str(self.minibatch_index).zfill(2)}/{str(self.minibatch_count - 1).zfill(2)}"
+        partner_id_str = f"Partner partner_id #{partner_id} ({partner_index}/{self.partners_count - 1})"
+        val_acc_str = f"{round(history['val_accuracy'][-1], 2)}"
 
-    def build_model_from_weights(self, new_weights):
-        """Generate a new model initialized with weights passed as arguments"""
-        new_model = self.generate_new_model()
-        new_model.set_weights(new_weights)
-        return new_model
+        logger.debug(f"{epoch_nb_str} > {mb_nb_str} > {partner_id_str} > val_acc: {val_acc_str}")
 
-    def init_model(self):
-        new_model = self.generate_new_model()
+    def log_model_val_perf(self):
```

Comment:
I would cut this method in two:

- `evaluate_val_perf() `which returns a hist object
- `log_perf(hist)`

Reply:
Also can we indicate it's local or global validation dataset ?

Reply:
My idea is that it already exists a evaluate method, the Keras one, so I prefer not duplicate it.

Reply:
I can add a comment, if you want. As we use the global method (mpl's one), it is computed on the global set. but maybe it is not so clear

Reply:
Other solution: rename this method to something like: `evaluate_on_val_data(`) ?

Reply:
Yes a comment would be a great for the global/local validation.

Reply:
What method ? the log ?

Reply:
If you want we can indicate in the method name that we are doing evaluation. But If someone wants to evaluate the model, he will go for the model.evaluate() method, with the data he wants to use. The log methods should never be called by an user, but only on purpose, during a home-build training (like in pvrl). The objective of this log methods is to centralized the history maintaining, and to provide an easy-to-use way of having proper logs and history while designing new contrib or mpl without having to worries about those. It kind of encapsulates the log handling. 

By cutting these methods in two, we would end up with evaluate_on_test_set() and evaluate_on_val_set(), which will be used only once by the logs methods.
Moreover these will be the equivalents of model.evaluate(val_data) and model.evaluate(test_data)

Therefore I think we should kept the method in one piece. 
But we can change their names if the 'log' is strange

Reply:
I agree to keep the method in one piece.
I think we should definitely change the name since the method is not only logging things. What about renaming the method to `evaluate_on_val_data() `? Any idea @natct10 @bowni ?

Reply:
> I agree to keep the method in one piece.
> I think we should definitely change the name since the method is not only logging things. What about renaming the method to `evaluate_on_val_data() `? Any idea @natct10 @bowni ?

It is logging:

```python
        if self.minibatch_index >= self.minibatch_count - 1:
            logger.info(f"   Model evaluation at the end of the epoch: "
                        f"{['%.3f' % elem for elem in hist]}")
```

Reply:
Ah okay, got it, so let's rename it as suggested